src/main/java/de/tum/cit/aet/artemis/core/service/feature/FeatureToggleService.java

```diff
@@ -36,10 +42,22 @@ public FeatureToggleService(WebsocketMessagingService websocketMessagingService,
         this.hazelcastInstance = hazelcastInstance;
     }
 
+    private Optional<Map<Feature, Boolean>> getFeatures() {
+        try {
+            if (hazelcastInstance != null && hazelcastInstance.getLifecycleService().isRunning()) {
+                return Optional.ofNullable(features);
+            }
+        }
+        catch (HazelcastInstanceNotActiveException e) {
+            log.error("Failed to get features in FeatureToggleService as Hazelcast instance is not active any more.");
```

Comment:
No super strong opinion on this, however, this might be a more robust logging implementation in case of refactorings?
```suggestion
            log.error("Failed to get features in {} as Hazelcast instance is not active anymore.", FeatureToggleService.class.getSimpleName());
```